concrete_datastore/api/v1/views.py

```diff
@@ -233,6 +235,53 @@ def options(self, request, *args, **kwargs):
         return Response(data=default_options, status=HTTP_200_OK)
 
 
+class RetrieveSecureConnectCode(generics.GenericAPIView):
+    serializer_class = ResetPasswordSerializer
+
+    def post(self, request, *args, **kwargs):
+        # The ResetPasswordSerializer only need an email like this view
+        serializer = ResetPasswordSerializer(data=request.data)
+        if not serializer.is_valid():
+            return Response(
+                data={
+                    "message": serializer.errors,
+                    "_errors": ["INVALID_DATA"],
+                },
+                status=HTTP_400_BAD_REQUEST,
+            )
+
+        email = serializer.validated_data["email"].lower()
+
+        user_queryset = UserModel.objects.filter(email=email, is_active=True)
+        if not user_queryset.exists():
+            return Response(
+                data={
+                    "message": "Wrong email address",
+                    "_errors": ["WRONG_EMAIL_ADDRESS"],
+                },
+                status=HTTP_400_BAD_REQUEST,
+            )
+
+        user = user_queryset.first()
+        # Check if the secure token is expired and expire in database
+        secure_codes = SecureConnectCode.objects.filter(
+            user=user, expired=False
+        )
+        for secure_code in secure_codes:
+            expire_secure_connect_instance(
+                secure_code, settings.SECURE_CONNECT_CODE_EXPIRY_TIME_SECONDS
+            )
+        secure_codes_count = secure_codes.count()
+        if secure_codes_count >= settings.MAX_SECURE_CONNECT_CODES:
```

Comment:
settings name not explicit enough

max  **simultaneously** active secure connect codes ?

Reply:
per user

Reply:
settings changed to `MAX_SIMULTANEOUS_SECURE_CONNECT_CODES_PER_USER`